src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -3541,6 +3555,9 @@ void BuildCustomMarkupExtensionPropertyValue(IIndentedStringBuilder writer, Xaml
 			}
 		}
 
+		private static bool IsNotFrameworkElementButNeedsSourceLocation(XamlObjectDefinition objectDefinition)
+			=> objectDefinition.Type.Name is "VisualState" or "AdaptiveTrigger" or "StateTrigger";
```

Comment:
We probably should be using lazy loaded symbols instead, like those, for performance reasons but also for exact type matching: 

https://github.com/unoplatform/uno/blob/1ddae936617e55845e1115482c057df92634a573/src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGeneration.cs#L260

Reply:
> We probably should be using lazy loaded symbols instead, like those, for performance reasons but also for exact type matching:

Is this a general comment or a specific request for a change?
Is the performance issue about comparing with strings? There are equivalent checks elsewhere in the file, should they be changed too?
Are there examples of other places which already do a better/preferred way of handling this?
How to measure the perf impact of this, and any change?

Reply:
Actually, I thought `.Type` was a roslyn symbol, but it's not in this case, since it's a string coming from the XAML parser. Let's keep it this way for now, apologies for the confusion.